test/functional/masternode_activation.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The PIVX developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import PivxTestFramework
+from test_framework.util import *
+import time
+
+"""
+Test checking:
+ 1) Masternode setup/creation.
+ 2) Tier two network sync (masternode broadcasting).
+ 3) Masternode activation.
+ 4) Masternode expiration.
+ 5) Masternode re activation.
+ 6) Masternode removal.
+ 7) Masternode collateral spent removal.
+"""
+
+class MasternodeActivationTest(PivxTestFramework):
+
+    def set_test_params(self):
+        self.num_nodes = 3
+        self.extra_args = [[], [], []]
+        self.setup_clean_chain = True
+
+    def check_mnsync_finished(self):
+        for i in range(0, 2):
+            status = self.nodes[i].mnsync("status")["RequestedMasternodeAssets"]
+            assert_equal(status, 999) # sync finished
+
+    def check_first_mn_in_mnlist_status(self, node, status):
+        listMNs = node.listmasternodes()
+        assert(len(listMNs) > 0)
+        assert_equal(listMNs[0]["status"], status)
+
+    def check_mn_list_empty(self, node):
+        assert(len(node.listmasternodes()) == 0)
+
+    def start_masternode(self, mnOwner, masternodeAlias, mnRemote, miner):
+        ret = mnOwner.startmasternode("alias", "false", masternodeAlias)
+        assert_equal(ret["result"], "success")
+
+        self.log.info("masternode started, waiting 125 seconds until it's enabled..")
+        for i in range(1, 5):
+            miner.generate(1)
+            sync_blocks(self.nodes)
+            time.sleep(25)
+
+        # check masternode enabled
+        self.check_first_mn_in_mnlist_status(mnRemote, "ENABLED")
+        self.check_first_mn_in_mnlist_status(mnOwner, "ENABLED")
+        self.check_first_mn_in_mnlist_status(miner, "ENABLED")
+
+    def start_remote_mn_and_sync_tier_two(self, miner):
+        self.start_node(1)
+        connect_nodes(self.nodes[0], 1)
+        connect_nodes(self.nodes[1], 0)
+        connect_nodes(self.nodes[1], 2)
+        connect_nodes(self.nodes[2], 1)
+
+        self.log.info("syncing tier two across recently started peers..")
+        # let the nodes sync the tier two
+        time.sleep(20)
+        miner.generate(1)
+        sync_blocks(self.nodes)
+
+        # wait a little bit until the tier two is synced.
+        time.sleep(20)
+        self.check_mnsync_finished()
+
+    def satoshi_round(amount):
+        return Decimal(amount).quantize(Decimal('0.00000001'), rounding=ROUND_DOWN)
```

Comment:
no need to duplicate/reimplement this. Just import it from util.py

Reply:
updating